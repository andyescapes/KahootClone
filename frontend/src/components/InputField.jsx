import React from 'react';
import TextField from '@material-ui/core/TextField';
import PropTypes from 'prop-types';

function InputField (props) {
  return (
    <div>
      <TextField
        onChange={(e) => props.setState(e.target.value)}
        label={props.field}
        value={props.state}
        type={props.type ?? 'text'}
        name={props.field}
      />
      {/* <input
        type = {props.type ?? "text"}
        onChange = {e=> props.setState(e.target.value)
        }>
      </input> */}
    </div>
  );
}
InputField.propTypes = {
  state: PropTypes.string,
  type: PropTypes.string,
  setState: PropTypes.func,
  field: PropTypes.string
}
export default InputField;
