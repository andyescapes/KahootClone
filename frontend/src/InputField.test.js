import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import InputField from './components/InputField';
import TextField from '@material-ui/core/TextField';

describe('InputField', () => {
  const noop = () => {};

  it('contains a Material UI Textfield', () => {
    const wrapper = shallow(<InputField />);
    expect(wrapper.find(TextField).length).toBe(1);
    console.log(wrapper.find(TextField));
  });

  it('TextField contains the value supplied via props', () => {
    const wrapper = shallow(<InputField state={'hello'} />);
    expect(wrapper.find(TextField).prop('value')).toBe('hello');
    console.log(wrapper.find(TextField));
  });
  it('TextField contains the correct label supplied via props', () => {
    const wrapper = shallow(<InputField field={'test'} />);
    expect(wrapper.find(TextField).prop('label')).toBe('test');
    console.log(wrapper.find(TextField));
  });

  it('triggers onChange event handler which calls setState when value changed', () => {
    const setState = jest.fn();

    const event = {
      target: {
        value: 'test value',
      },
    };
    const wrapper = shallow(<InputField setState={setState} />);
    wrapper.find(TextField).at(0).simulate('change', event);
    expect(setState).toHaveBeenCalledTimes(1);
  });

  it('renders with minimal props', () => {
    const textField = renderer.create(<TextField setState={noop} />).toJSON();
    expect(textField).toMatchSnapshot();
  });

  it('renders with with provided label (snapshot)', () => {
    const textField = renderer
      .create(<TextField setState={noop} field={'a title'} />)
      .toJSON();
    expect(textField).toMatchSnapshot();
  });
  it('renders with with provided value (snapshot)', () => {
    const textField = renderer
      .create(<TextField setState={noop} state={'my value'} />)
      .toJSON();
    expect(textField).toMatchSnapshot();
  });
});
