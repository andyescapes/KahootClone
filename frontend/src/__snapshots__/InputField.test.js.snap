// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders with minimal props 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
  setState={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;

exports[`InputField renders with with provided label (snapshot) 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
  field="a title"
  setState={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;

exports[`InputField renders with with provided value (snapshot) 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
  setState={[Function]}
  state="my value"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;
